hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReaderSchemaHandler.java

```diff
@@ -72,22 +71,21 @@ public class HoodieFileGroupReaderSchemaHandler<T> {
 
   protected final boolean needsMORMerge;
 
-  public HoodieFileGroupReaderSchemaHandler(HoodieReaderContext<T> readerContext,
+  public HoodieFileGroupReaderSchemaHandler(FileGroupReaderState readerState,
                                             Schema dataSchema,
                                             Schema requestedSchema,
-                                            Option<InternalSchema> internalSchemaOpt,
-                                            HoodieTableConfig hoodieTableConfig) {
-    this.readerContext = readerContext;
-    this.hasBootstrapBaseFile = readerContext.getHasBootstrapBaseFile();
-    this.needsMORMerge = readerContext.getHasLogFiles();
-    this.recordMerger = readerContext.getRecordMerger();
+                                            Option<InternalSchema> internalSchemaOpt) {
+    this.readerState = readerState;
+    this.hasBootstrapBaseFile = readerState.getHasBootstrapBaseFile();
+    this.needsMORMerge = readerState.getHasLogFiles();
+    this.recordMerger = readerState.getRecordMerger();
```

Comment:
Similar here.  Remove all variables that can be replaced by `readerState` calls.